From 55c1bc61d284a0835c2dc7de65ee44505bd7f5a8 Mon Sep 17 00:00:00 2001
From: Marius Bagassien <marius.b@variscite.com>
Date: Thu, 30 Apr 2026 02:17:53 +0200
Subject: [PATCH 1/2] variscite: bootanimation: add build-time logo override

Allow products to substitute the android() boot animation logo
textures via two soong config variables:
  bootanimation.logo_mask
  bootanimation.logo_shine
Falls back to the stock Google assets when not set.

Change-Id: I8975e443f1bd6b269c118758c276d59462d1fcf8
Signed-off-by: Marius Bagassien <marius.b@variscite.com>
---
 cmds/bootanimation/Android.bp        | 30 +++++++++++++++++++++++++++-
 cmds/bootanimation/BootAnimation.cpp | 13 ++++++++++--
 2 files changed, 40 insertions(+), 3 deletions(-)

diff --git a/cmds/bootanimation/Android.bp b/cmds/bootanimation/Android.bp
index f80ccf7a56fb..c95ff9a96e45 100644
--- a/cmds/bootanimation/Android.bp
+++ b/cmds/bootanimation/Android.bp
@@ -1,3 +1,28 @@
+// Build-time boot-animation logo override. Set via soong_config_set
+// in a product .mk: bootanimation.logo_mask / bootanimation.logo_shine.
+soong_config_module_type {
+    name: "bootanimation_logo_cc_defaults",
+    module_type: "cc_defaults",
+    config_namespace: "bootanimation",
+    value_variables: [
+        "logo_mask",
+        "logo_shine",
+    ],
+    properties: ["cflags"],
+}
+
+bootanimation_logo_cc_defaults {
+    name: "bootanimation_logo_defaults",
+    soong_config_variables: {
+        logo_mask: {
+            cflags: ["-DBOOT_LOGO_MASK=\"%s\""],
+        },
+        logo_shine: {
+            cflags: ["-DBOOT_LOGO_SHINE=\"%s\""],
+        },
+    },
+}
+
 package {
     // See: http://go/android-license-faq
     // A large-scale-change added 'default_applicable_licenses' to import
@@ -79,7 +104,10 @@ cc_binary {
 
 cc_library_shared {
     name: "libbootanimation",
-    defaults: ["bootanimation_defaults"],
+    defaults: [
+        "bootanimation_defaults",
+        "bootanimation_logo_defaults",
+    ],
 
     srcs: ["BootAnimation.cpp"],
 
diff --git a/cmds/bootanimation/BootAnimation.cpp b/cmds/bootanimation/BootAnimation.cpp
index b0070c5faa36..575c78d61919 100644
--- a/cmds/bootanimation/BootAnimation.cpp
+++ b/cmds/bootanimation/BootAnimation.cpp
@@ -110,6 +110,15 @@ static const int TEXT_MISSING_VALUE = INT_MIN;
 static const char EXIT_PROP_NAME[] = "service.bootanim.exit";
 static const char PROGRESS_PROP_NAME[] = "service.bootanim.progress";
 static const char CLOCK_ENABLED_PROP_NAME[] = "persist.sys.bootanim.clock.enabled";
+// Boot animation logo paths. Override at build time via the
+// PRODUCT_BOOTANIMATION_LOGO_{MASK,SHINE} product variables.
+#ifndef BOOT_LOGO_MASK
+#define BOOT_LOGO_MASK "images/android-logo-mask.png"
+#endif
+#ifndef BOOT_LOGO_SHINE
+#define BOOT_LOGO_SHINE "images/android-logo-shine.png"
+#endif
+
 static const int ANIM_ENTRY_NAME_MAX = ANIM_PATH_MAX + 1;
 static const int MAX_CHECK_EXIT_INTERVAL_US = 50000;
 static constexpr size_t TEXT_POS_LEN_MAX = 16;
@@ -845,8 +854,8 @@ bool BootAnimation::android(const Display& display) {
 
     SLOGD("%sAnimationShownTiming start time: %" PRId64 "ms", mShuttingDown ? "Shutdown" : "Boot",
             elapsedRealtime());
-    initTexture(&mAndroid[0], mAssets, "images/android-logo-mask.png");
-    initTexture(&mAndroid[1], mAssets, "images/android-logo-shine.png");
+    initTexture(&mAndroid[0], mAssets, BOOT_LOGO_MASK);
+    initTexture(&mAndroid[1], mAssets, BOOT_LOGO_SHINE);
 
     mCallbacks->init({});
 
-- 
2.43.0

